### PR TITLE
DESENG-475 Add contributing guidelines for repo

### PR DESCRIPTION
**Pull request by Baelx**
Issue #: https://apps.itsm.gov.bc.ca/jira/browse/DESENG-475

Add contributing guideline to repository.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -4,6 +4,59 @@ Government employees, public and members of the private sector are encouraged to
...
+
+With this method of segmentation, you don't need to worry about committing incomplete or broken features to main. When all child branches have been completed for the feature, the parent is then merged back into main in a final PR. This final review can contain your CHANGELOG.md entry. In the title or description of your PR, please indicate that all work has already been reviewed.
+
+## Merge Techniques
+
+Due to how PRs are commonly segmented in this repository, and due to its active nature, rebasing is discouraged as a way to merge changes. Traditional merge commits are preferred as they preserve history and make it easier to resolve merge conflicts.
+
+## Guidelines for reviewers
+
+Conditions for whether to Approve or Request Changes for a PR can come with a lot of nuance. We try to air on the side of Approve unless there are obvious changes needed. Below is a list of reasons to "stop" a PR. Items with a star next to them are subject to interpretation by the review. Some items may prompt to discussion between the reviewer and the submitting developer so that choices can be justified.
```
Comment (NatSquared) [inline CONTRIBUTING.md:47]: "err on the side of"?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -4,6 +4,59 @@ Government employees, public and members of the private sector are encouraged to
...
+
+Examples of when to Request Changes
+- Logging functions are left in that aren't required or needed later
```
Comment (NatSquared) [inline CONTRIBUTING.md:50]: Perhaps carve out an explicit exception for `console.error` if we're going forward with that

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -4,6 +4,59 @@ Government employees, public and members of the private sector are encouraged to
...
+- Commented-code is left in that isn't needed later
+- Widespread incorrect syntax use (using snake case for many variables in JS when it's not required)
+- Significant amount of repeition in code that can be condensed without much issue (code isn't DRY) *
+- Unnecessary function calls, overly verbose code *
+
+## Common Components/Utilities
+
+- `@transational`: Database method decorator. If there is an exception during execution, the entire DB session will be safely rolled back to a point in time just before the decorated function was called. If not, the session will be saved, unless autocommit is set to False. This helps replace most session management boilerplate.
+- `authorization.check_auth`: Checks a user for one or more roles and for the correct engagement.
```
Comment (NatSquared) [inline CONTRIBUTING.md:59]: These are really helpful to have documented. I will definitely be adding to this section in the future :)

File: docs/pr-segmentation.png
Comment (NatSquared) [inline docs/pr-segmentation.png:1]: Slick diagram!

**Review by NatSquared — APPROVED** [review]
LGTM :) Thanks for helping us get organized

File: CONTRIBUTING.md (hunk 3 above)
Comment (Baelx) [reply CONTRIBUTING.md:59]: Awesome! I'm hoping that having a reference list of common utilities could speed up development in the future

**Pull request merged**